### PR TITLE
Added end-to-end test cases for checking the healthy states of all agent nodes

### PR DESCRIPTION
**Pull request by soccerGB**
A new test, AreAllNodesHealthy, was added to the existing scaler unitests for validating the agent nodes healthy in the following cases:

1. Initial cluster deployment
2. after cluster scale-up operation
3. after cluster scale-down operation

Note: This PR along with the follow two PRs need to be merged for the same deployment to avoid test failures.

https://github.com/Microsoft/mesos-jenkins/pull/37
https://github.com/dcos/dcos-windows/pull/24

### CODE REVIEW

File: Diagnostics/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+## Mesos Windows agent building and testing
```
Comment (ionutbalutoiu) [inline Diagnostics/README.md]: This is the `Mesos` `README`. Please make a simple `README` for Diagnostics or simply remove it and we'll add later.

File: Diagnostics/start-windows-build.ps1 [hunk 2]
```diff
@@ -0,0 +1,482 @@
+Param(
```
Comment (ionutbalutoiu) [inline Diagnostics/start-windows-build.ps1]: I think that this change is added to the PR https://github.com/Microsoft/mesos-jenkins/pull/37 and should not be included here.

File: Diagnostics/utils/common.py [hunk 3]
```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python
```
Comment (ionutbalutoiu) [inline Diagnostics/utils/common.py]: We don't need this file here.

File: Diagnostics/utils/get-review-ids.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python
```
Comment (ionutbalutoiu) [inline Diagnostics/utils/get-review-ids.py]: We don't need this file here.

File: Diagnostics/utils/jenkins-executor-common.sh [hunk 5]
```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env bash
```
Comment (ionutbalutoiu) [inline Diagnostics/utils/jenkins-executor-common.sh]: We don't need this file here.

File: Diagnostics/utils/post-build-result.py [hunk 6]
```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
```
Comment (ionutbalutoiu) [inline Diagnostics/utils/post-build-result.py]: We don't need this file here.

File: Diagnostics/utils/trigger-gearman-jobs.py [hunk 7]
```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python
```
Comment (ionutbalutoiu) [inline Diagnostics/utils/trigger-gearman-jobs.py]: We don't need this file here.

File: Diagnostics/utils/verify-review-requests.py [hunk 8]
```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python
```
Comment (ionutbalutoiu) [inline Diagnostics/utils/verify-review-requests.py]: We don't need this file here.

File: Mesos/start-windows-build.ps1 [hunk 9]
```diff
@@ -35,7 +35,7 @@ function Install-Prerequisites {
             'install_args' = @("/SILENT")
             'install_dir' = $GIT_DIR
         }
-        'cmake'= @{
+        'go'= @{
```
Comment (ionutbalutoiu) [inline Mesos/start-windows-build.ps1]: Why does exactly you do this change ?

File: global-variables.ps1 [hunk 10]
```diff
@@ -50,6 +50,23 @@ $SPARTAN_BUILD_OUT_DIR = Join-Path $SPARTAN_DIR "build-output"
 $SPARTAN_BUILD_LOGS_DIR = Join-Path $SPARTAN_BUILD_OUT_DIR "logs"
 $SPARTAN_BUILD_BASE_URL = "$LOG_SERVER_BASE_URL/spartan-build"
 
+# Diagnostics configurations
```
Comment (ionutbalutoiu) [inline global-variables.ps1]: These additions are already included in https://github.com/Microsoft/mesos-jenkins/pull/37

File: Diagnostics/README.md (hunk 1 above)
Comment (soccerGB) [reply Diagnostics/README.md]: Agree they shouldn't be included.
I am trying to find out why those those extra files were added in this PR.

Comment (soccerGB) [reply Diagnostics/README.md]: I will remove it

File: Diagnostics/start-windows-build.ps1 (hunk 2 above)
Comment (soccerGB) [reply Diagnostics/start-windows-build.ps1]: I will remove it

File: Diagnostics/utils/common.py (hunk 3 above)
Comment (soccerGB) [reply Diagnostics/utils/common.py]: I will remove it

File: Diagnostics/utils/get-review-ids.py (hunk 4 above)
Comment (soccerGB) [reply Diagnostics/utils/get-review-ids.py]: I will remove it

File: Diagnostics/utils/jenkins-executor-common.sh (hunk 5 above)
Comment (soccerGB) [reply Diagnostics/utils/jenkins-executor-common.sh]: I will remove it

File: Diagnostics/utils/post-build-result.py (hunk 6 above)
Comment (soccerGB) [reply Diagnostics/utils/post-build-result.py]: I will remove it

File: Diagnostics/utils/trigger-gearman-jobs.py (hunk 7 above)
Comment (soccerGB) [reply Diagnostics/utils/trigger-gearman-jobs.py]: I will remove it

File: Diagnostics/utils/verify-review-requests.py (hunk 8 above)
Comment (soccerGB) [reply Diagnostics/utils/verify-review-requests.py]: I will remove it

File: global-variables.ps1 (hunk 10 above)
Comment (soccerGB) [reply global-variables.ps1]: I will remove it

File: Mesos/start-windows-build.ps1 (hunk 9 above)
Comment (soccerGB) [reply Mesos/start-windows-build.ps1]: It looks like a bad merge was causing those extra files and this unintended change to be included. I am trying to address this.

**Comment by soccerGB** [comment]
I made a mistake when creating this "health" branch, instead of branching out from the master, I did it from a "diagnostics" branch that has other unwanted files and changes. Sorry for the PR messup. It's all good now

**Comment by ionutbalutoiu** [comment]
Thank-you! I'll merge this after I test it in the CI.

Ionut

**Pull request merged**